BR: reorganize content about BR tool (#4810)
pingcap/docs: br/backup-and-restore-use-cases.md

# BR Use Cases

[BR](/br/backup-and-restore-tool.md) is a tool for distributed backup and restoration of the TiDB cluster data.

> This document describes how to runn BR in the following use cases:

Author: ```suggestion
This document describes how to run BR in the following use cases:
```

## Use cases

This document describes the following use cases:

* [Back up a single table to a network disk (recommended in production environment)](#back-up-a-single-table-to-a-network-disk-recommended-in-production-environment)
* [Restore data from a network disk (recommended in production environment)](#restore-data-from-a-network-disk-recommended-in-production-environment)
* [Back up a single table to a local disk (recommended in testing environment)](#back-up-a-single-table-to-a-local-disk-recommended-in-testing-environment)
* [Restore data from a local disk (recommended in testing environment)](#restore-data-from-a-local-disk-recommended-in-testing-environment)

It is recommended that you use a network disk to back up and restore data. This spares you from collecting backup files and greatly improves the backup efficiency especially when the TiKV cluster is in a large scale.

Before the backup or restoration operations, you need to do some preparations:

- [Preparation for backup](#preparation-for-backup)
- [Preparation for restoration](#preparation-for-restoration)

### Preparation for backup

> In TiDB v4.0.8 and higher versions, BR supports the self-adaptive Garbage Collection (GC). So to avoid manually configuring GC, you only need to register `backupTS` in `safePoint` in PD and make sure that `safePoint` does not move forward during the backup process.

Author: ```suggestion
In TiDB v4.0.8 and later versions, BR supports the self-adaptive Garbage Collection (GC). So to avoid manually configuring GC, you only need to register `backupTS` in `safePoint` in PD and make sure that `safePoint` does not move forward during the backup process.
```

> In TiDB v4.0.7 and lower versions, you need to manually configure GC before and after the BR backup through the following steps:

Author: ```suggestion
In TiDB v4.0.7 and earlier versions, you need to manually configure GC before and after the BR backup through the following steps:
```

* The throughput of a single TiKV instance: `avg speed(MB/s)`/`tikv_count` = `97.2`
* The average restoration speed of a single TiKV instance: `total size(MB)`/(`split time` + `restore time`)/`tikv_count` = `92.4`

## Error handling during backup

This section introduces the common errors occurred during the backup process.

### `key locked Error` in the backup log

Error message in the log: `log - ["backup occur kv error"][error="{\"KvError\":{\"locked\":`

> If a key is locked during the backup process, BR tries to resolve the lock. A small number of this error does not affect the correctness of the backup.

Author: ```suggestion
If a key is locked during the backup process, BR tries to resolve the lock. A small number of these errors do not affect the correctness of the backup.
```
